File: app/src/main/java/com/vereshchagin/nikolay/stankinschedule/modulejournal/network/response/SemestersResponse.java
package com.vereshchagin.nikolay.stankinschedule.modulejournal.network.response;

import androidx.annotation.NonNull;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

/**
 * POJO-класс "semesters" ответа от сервара.
 */
public class SemestersResponse {

    @SerializedName("surname")
    @Expose
    private String mSurname;

    @SerializedName("initials")
    @Expose
    private String mInitials;

    @SerializedName("stgroup")
    @Expose
    private String mGroup;

    @SerializedName("semesters")
    @Expose
    private ArrayList<String> mSemesters;

    @NonNull
    public String studentName() {
        return mSurname + " " + mInitials;
    }

    @NonNull
    public String group() {
        return mGroup;
    }

    @NonNull
    public ArrayList<String> semesters() {
        return mSemesters;
    }

    @NonNull
    @Override
    public String toString() {
        return String.format("[%s : %s : %s]", studentName(), group(), semesters());
    }
}

File: app/src/main/java/com/vereshchagin/nikolay/stankinschedule/model/modulejournal/SemestersResponse.kt
package com.vereshchagin.nikolay.stankinschedule.model.modulejournal

import com.google.gson.annotations.SerializedName

/**
 * Ответ с информацией о студенте и его семестрах от сервера.
 */
class SemestersResponse(
    @SerializedName("surname")
    val surname: String,
    @SerializedName("initials")
    val initials: String,
    @SerializedName("stgroup")
    val group: String,
    @SerializedName("semesters")
    val semesters: List<String>
) {
    val fullname get() = "$surname $initials"

    override fun toString(): String {
        return "SemestersResponse(surname='$surname', initials='$initials', group='$group', semesters=$semesters)"
    }
}